### ISSUE TITLE
Getting Segmentation fault

### ISSUE THREAD

**Issue by brijesh-6899** [grammarly/gector#37]
Hi,

I've been trying to use the trained model (roberta_1_gector.th) on a GPU instance using this query _"python3.6 predict.py --model_path models/roberta_1_gector.th --vocab_path data/output_vocabulary/ --input_file ../ data/ip.txt --output_file ../op.txt --min_error_probability 0.2 --additional_confidence 0.5"_

But it is giving me this error trace:


Fatal Python error: Segmentation fault                                                                                                                                                                                                                                                                                                                                                            
Current thread 0x00007fbdf39c9700 (most recent call first):                                                                                                                                        
File "/data/home/ubuntu/brijesh_intern/cqi_pip_venv/lib/python3.6/site-packages/torch/nn/modules/sparse.py", line 97 in __init__                                                                 
File "/data/home/ubuntu/brijesh_intern/cqi_pip_venv/lib/python3.6/site-packages/transformers/modeling_bert.py", line 160 in __init__                                                            
File "/data/home/ubuntu/brijesh_intern/cqi_pip_venv/lib/python3.6/site-packages/transformers/modeling_bert.py", line 617 in __init__                                                             
File "/data/home/ubuntu/brijesh_intern/cqi_pip_venv/lib/python3.6/site-packages/transformers/modeling_roberta.py", line 181 in __init__                                                          
File "/data/home/ubuntu/brijesh_intern/cqi_pip_venv/lib/python3.6/site-packages/transformers/modeling_utils.py", line 406 in from_pretrained                                                     
File "/data/home/ubuntu/brijesh_intern/cqi_pip_venv/lib/python3.6/site-packages/transformers/modeling_auto.py", line 156 in from_pretrained                                                      
File "/data/home/ubuntu/brijesh_intern/grammarly_gector/gector/gector/bert_token_embedder.py", line 29 in load                                                                                   
File "/data/home/ubuntu/brijesh_intern/grammarly_gector/gector/gector/bert_token_embedder.py", line 254 in __init__                                                                              
File "/data/home/ubuntu/brijesh_intern/grammarly_gector/gector/gector/gec_model.py", line 176 in _get_embbeder                                                                                   
File "/data/home/ubuntu/brijesh_intern/grammarly_gector/gector/gector/gec_model.py", line 88 in __init__                                                                                         
File "predict.py", line 46 in main                                                                                                                                                               
File "predict.py", line 120 in                                                                                                                                                         
Segmentation fault

**Comment by komelianchuk** [comment]
Hi.
What is the size of the input file?
Please check that GPU is indeed used and consider using a smaller batch size during the prediction.

**Comment by brijesh-6899** [comment]
@komelianchuk Thanks for the reply!

Input file "ip.txt" is having 4-5 lines only as of now! 
I tried with reduced batch_size (=64)  but it's throwing the same error. For GPU usage, I tried looking into the nvidia-smi output, it displays no GPU memory consumption and in fact, after hitting the run command, programs exits within 1-2 seconds!

**Comment by komelianchuk** [comment]
I can suggest debugging this problem more closely.
It seems like you can not load the model. How much memory do you have on your GPU? Can you load the model through python console? Can you try to run this code on CPU?

**Comment by brijesh-6899** [comment]
Sorry, Forgot to mention, I tried it running on machine w/o GPU and there it worked, on CPU! 
I tried loading it using the python console, but it failed with the same error, It failed during line 31 of predict.py!
I'm using Nvidia Tesla K80 GPU with 12GB of Memory.

**Comment by komelianchuk** [comment]
Please try to update CUDA to the latest version.

**Comment by brijesh-6899** [comment]
updated CUDA to latest version 11 and tried executing the same, but still, it is giving the segmentation fault error!

**Comment by komelianchuk** [comment]
Something is definitely wrong with your GPU. It should be enough memory to load the model, so it seems like you have some software problem. (multiple CUDA installed, wrong paths, old drivers, etc)
It's hard to say what exactly wrong.
Consider running some basic transformers examples in order to verify that.

```
>>> from transformers import BertTokenizer, BertModel
>>> import torch

>>> tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
>>> model = BertModel.from_pretrained('bert-base-uncased')

>>> inputs = tokenizer("Hello, my dog is cute", return_tensors="pt")
>>> outputs = model(**inputs)

>>> last_hidden_states = outputs[0]
```

**Comment by brijesh-6899** [comment]
I tried running the above code snippet, but it failed with the same error. Then I looked through Hagging Face's and PyTorch's issue pages and found this reference/solution:

- Changed sentencepiece to version 0.1.91 from 0.1.92 ([issue page](https://github.com/huggingface/transformers/issues/5281)) and it worked,

Prior to this, I did some other changes as well:
1. Updated CUDA version from 10.1 to 11.0
2. Changed Scipy version from 1.5 to 1.4.1 ([issue page](https://github.com/huggingface/transformers/pull/2207)) 

@komelianchuk Thanks a lot for helping!
